internal/factsengine/gatherers/saplocalhostresolver.go

```diff
@@ -0,0 +1,146 @@
+package gatherers
+
+import (
+	"net"
+	"path/filepath"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/trento-project/agent/internal/core/sapsystem"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const SapLocalhostResolverGathererName = "saplocalhost_resolver"
+
+// nolint:gochecknoglobals
+var (
+	SapLocalhostResolverHostnameResolutionError = entities.FactGatheringError{
+		Type:    "saplocalhost_resolver-resolution-error",
+		Message: "error resolving hostname",
+	}
+)
+
+// saplocalhost_reachability
+type SapLocalhostResolverGatherer struct {
+	fs afero.Fs
+	hr utils.HostnameResolver
+}
+
+func NewDefaultSapLocalhostResolverGatherer() *SapLocalhostResolverGatherer {
+	return NewSapLocalhostResolver(afero.NewOsFs(), utils.Resolver{})
+}
+
+func NewSapLocalhostResolver(fs afero.Fs, hr utils.HostnameResolver) *SapLocalhostResolverGatherer {
+	return &SapLocalhostResolverGatherer{fs: fs, hr: hr}
+}
+
+func (r *SapLocalhostResolverGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	var facts []entities.Fact
+	profiles, _ := r.getProfiles()
+
+	for _, factReq := range factsRequests {
+		var fact entities.Fact
+		for _, profile := range profiles {
+			ips, err := r.hr.LookupHost(profile["SAPLOCALHOST"])
+			if err != nil {
+				log.Error(err)
+				gatheringError := SapLocalhostResolverHostnameResolutionError.Wrap(profile["SAPLOCALHOST"])
+				fact = entities.NewFactGatheredWithError(factReq, gatheringError)
+			} else {
+				ipList := &entities.FactValueList{}
+				for _, ip := range ips {
+					ipList.Value = append(ipList.Value, &entities.FactValueString{Value: ip})
+				}
+
+				// reachable := false
+				// for _, ip := range ips {
+				// 	if ok, _ := isHostReachable(ip); ok {
+				// 		reachable = true
+				// 		break
+				// 	}
+				// }
+
+				factValue := &entities.FactValueMap{
```

Comment:
You don't really need to create the `FactValueMap` yourself.
You can create your own internal struct with its own json annotations, and later on marshall/unmarshall/convertToFactValue as we do in many other gatherers